### PR TITLE
feat(CEECORE-3010): add a basic react v18 based subapp version 1 

### PR DESCRIPTION
**Pull request by mristic505**
### Description

- The SubApp can be found in `sample/subapp1-online-store` directory.
- Its goal is to demonstrate basic `ServerSide rendering`, `Stream`, `Redux implementation` and `Suspense` features.

- **SubApp Header** is an `SSR` subapp with a static Hello World message.
- **SubApp Main Body** demonstrates a working `Redux` example as well as `SSR` and `Stream` features.
- **SubApp Footer** demonstrates `Suspense` feature.

### Completed

- [x] Add basic react v18 based SubApp that uses Version 1 packages.
- [x] Upgrade Version 1 packages
- [x] Create 3 SubApps (Header, Main and Footer).
- [x] Demonstrate Redux implementation
- [x] Refactor [framework-lib.js](https://github.com/electrode-io/electrode/blob/b5d9c9bbccf86d6a80b4fe75d4aa89b6d56a983a/packages/subapp-react/lib/framework-lib.js) to work without `renderToStaticMarkup`.
- [x] Refactor [framework-lib.js](https://github.com/electrode-io/electrode/blob/b5d9c9bbccf86d6a80b4fe75d4aa89b6d56a983a/packages/subapp-react/lib/framework-lib.js) and [ssr-framework.spec.js](https://github.com/electrode-io/electrode/blob/c2d0ddd435b89ba2e02fabbe72e7158cd1117211/packages/subapp-react/test/spec/ssr-framework.spec.js) to work with `react-router` v6.
- [x] Develop working `Suspense` demo

### CODE REVIEW

File: samples/subapp1-online-store/package.json [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+      "@xarc/app": "../../packages/xarc-app",
+      "subapp-react": "../../packages/subapp-react",
+      "subapp-redux": "../../packages/subapp-redux",
+      "subapp-server": "../../packages/subapp-server",
+      "subapp-web": "../../packages/subapp-web"
+    },
+    "devDependencies": {
+      "@xarc/app-dev": "../../packages/xarc-app-dev",
+      "@xarc/opt-eslint": "../../packages/xarc-opt-eslint"
+    }
```
Comment (coston) [inline samples/subapp1-online-store/package.json:67]: nit: Does `fun bootstrap` do this for us?

File: samples/subapp1-online-store/package.json [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  },
+  "prettier": {
+    "printWidth": 100,
+    "trailingComma": "none",
+    "arrowParens": "avoid"
+  }
```
Comment (coston) [inline samples/subapp1-online-store/package.json:73]: nit: I recommend just sticking with the prettier defaults so the style does not change example to example.

File: samples/subapp1-online-store/src/footer/subapp-footer.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
+/* eslint-disable no-magic-numbers */
+
```
Comment (coston) [inline samples/subapp1-online-store/src/footer/subapp-footer.js:2]:
```suggestion
```

You could declare a variable for 2500 instead.

File: samples/subapp1-online-store/src/footer/subapp-footer.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+const Data1 = () => {
+  if (!data) {
+    throw new Promise(resolve => {
+      setTimeout(() => {
+        data = "demo data received from async call";
+        resolve(data);
+      }, 2500);
+    });
+  }
+  return <div>{data}</div>;
```
Comment (coston) [inline samples/subapp1-online-store/src/footer/subapp-footer.js:16]: Major: This does not seem to work. I recommend use a similar approach with Timeout and React.Lazy, but ditch reassigning the data1 value.

File: samples/subapp1-online-store/src/server-routes/default/index.js [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+          <Require
+            _id="subapp-web/lib/load"
+            name="Header"
+            _concurrent
+            timestamp
+            startOnLoad
+            elementId="subapp-header"
+            serverSideRendering
+            hydrateServerData={false}
+            clientSideRendering={false}
```
Comment (coston) [inline samples/subapp1-online-store/src/server-routes/default/index.js:48]:
```suggestion
```

**Review by coston — CHANGES_REQUESTED** [review]
Good progress, but it's not ready yet. Please consider my requested changes.

File: samples/subapp1-online-store/package.json (hunk 2 above)
Comment (mristic505) [reply samples/subapp1-online-store/package.json:73]: I think that `"printWidth": 100` is the only requirement according to this https://github.com/electrode-io/electrode/blob/master/CONTRIBUTING.md

I'll remove `trailingComma ` and `arrowParens`

File: samples/subapp1-online-store/package.json (hunk 1 above)
Comment (mristic505) [reply samples/subapp1-online-store/package.json:67]: Many samples use this approach by adding packages manually to FYN object. I am investigating options for linking dependencies with FYN.

Comment (coston) [reply samples/subapp1-online-store/package.json:67]: Thanks! I've seen that this fyn deps object is the common approach. Normally monorepo managers manage dependency linking without having to re-declare dependencies, so please share if you see that there's a better way with fyn.

**Review by smuthya — APPROVED** [review]
LGTM

**Pull request merged**